Components/LearningCore/Source/gov/sandia/cognition/statistics/distribution/GammaDistribution.java

```diff
@@ -360,86 +354,120 @@ public static void sampleInto(
         final int start,
         final int length)
     {
-        ArgumentChecker.assertIsPositive("shape", shape);
-        ArgumentChecker.assertIsPositive("scale", scale);
-
-        // Note: This code is duplicated in sampleAsDouble to avoid overhead
-        // of creating objects.
-        final int k = (int) Math.floor(shape);
-        final double delta = shape - k;
-        final double v0 = (delta > 0.0) ? Math.E / (Math.E + delta) : 0.0;
         final int end = start + length;
         for (int i = start; i < end; i++)
         {
-            output[i] = sample(random, shape, scale, k, delta, v0);
+            output[i] = sample(shape, scale, random);
         }
     }
     
-    // Internal sampling function implementation.
-    private static double sample(
-        final Random random,
+    /**
+     * Provides a single sample from a Gamma distribution with the given shape
+     * and scale.
+     * 
+     * @param shape
+     *      Shape parameter of the Gamma distribution, often written as "k",
+     *      must be greater than zero.
+     * @param scale
+     *      Scale parameters of the Gamma distribution, often written as "theta",
+     *      must be greater than zero.
+     * @param random
+     *      Random number generator to use.
+     * @return 
+     *      A value sampled from a Gamma distribution.
+     */
+    public static double sample(
         final double shape,
         final double scale,
-        final int k,
-        final double delta,
-        final double v0)
+        final Random random)
+    {
+        // Shape is checked in the next function.
+        ArgumentChecker.assertIsPositive("scale", scale);
+        return scale * sampleStandard(shape, random);
+    }
+    
+    /**
+     * Provides a single sample from a Gamma distribution with the given shape
+     * and a scale of 1.
+     * 
+     * @param shape
+     *      Shape parameter of the Gamma distribution, often written as "k",
+     *      must be greater than zero.
+     * @param random
+     *      Random number generator to use.
+     * @return 
+     *      A value sampled from a Gamma distribution.
+     */
+    public static double sampleStandard(
+        final double shape,
+        final Random random)
     {
-        double logSum = 0.0;
-        for( int i = 0; i < k; i++ )
+        ArgumentChecker.assertIsPositive("shape", shape);
+        
+        // This is based on the gamma distribution algorithm used in numpy.
```

Comment:
minor: add a link to distributions.c? https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/distributions.c